### ISSUE TITLE
ERR_HTTP_HEADERS_SENT Cannot set headers after they are sent to the client

### ISSUE THREAD

**Issue by felquis** [i18next/i18next-express-middleware#143]
I'm getting this error, I think it is related to this module,  I used the example from react-i18next, with nextjs.

I'm using this detection config:
```js
    detection: {
      order: ['querystring', 'cookie', 'header'],
      lookupQuerystring: 'lng',
      lookupCookie: 'lng',
      lookupSession: 'lng',
      caches: ['cookie'],
    }
```

My browser is in pt-BR, and I request the page with ?lng=en, I think it sets up the pt-BR language, then it tries to set to en and the server crashes with this error.

I can  post the entire server.js if it would help.

**Comment by jamuhl** [comment]
did you use the https://github.com/i18next/i18next-express-middleware#language-detection !?!

full server.js might help...

**Comment by yasincanakmehmet** [comment]
I had the same issue with v1.0.9 but with v1.0.10 it should be solved try to update your version. If it still doesn't work, it should :), then you can fix it by checking if the res._headerSent is false before calling the cacheUserLanguage.

**Comment by yasincanakmehmet** [comment]
I did a quickfix by checking if the headers are set before setting the cookies. I thought it was fixed with 1.0.10 but I realized I had the cache set to false :) I will create a PR for the fix as soon as I can.
